iree/hal/local/executable_library.h

```diff
@@ -82,33 +139,47 @@ typedef union {
   uint32_t value[3];
 } iree_hal_vec3_t;
 
-#if defined(_MSC_VER)
-typedef __declspec(
-    align(16)) const uint32_t* iree_hal_executable_push_constants_ptr_t;
-#else
-typedef const uint32_t* iree_hal_executable_push_constants_ptr_t
-    __attribute__((align_value(16)));
-#endif  // MSVC
-
-typedef void* iree_hal_executable_binding_ptr_t;
+// Read-only per-dispatch state passed to each workgroup in a dispatch.
+typedef struct {
+  // Total workgroup count for the dispatch. This is sourced from either the
+  // original dispatch call (for iree_hal_command_buffer_dispatch) or the
+  // indirection buffer (for iree_hal_command_buffer_dispatch_indirect).
+  iree_hal_vec3_t workgroup_count;
+  // Workgroup size chosen for the dispatch. For compilation modes where the
+  // workgroup size is constant this may be ignored.
+  iree_hal_vec3_t workgroup_size;
+
+  // Total number of available 4 byte push constant values in |push_constants|.
+  size_t push_constant_count;
+  // |push_constant_count| values.
+  const uint32_t* push_constants;
+
+  // Total number of binding base pointers in |binding_ptrs| and
+  // |binding_lengths|. The set is packed densely based on which binidngs are
+  // used (known at compile-time).
+  size_t binding_count;
+  // Base pointers to each binding buffer.
+  void* const* binding_ptrs;
+  // The length of each binding in bytes, 1:1 with |binding_ptrs|.
+  const size_t* binding_lengths;
+
+  // Optional imported functions available for use within the executable.
+  const iree_hal_executable_import_table_v0_t* imports;
```

Comment:
NEAT!

Reply:
Still needs to be defined/requires goo - but at least there's a placeholder I can point to when people ask where it would go :)